fastestimator/pipeline.py

```diff
@@ -519,3 +551,57 @@ def _pad_batch_collate(self, batch: List[MutableMapping[str, Any]]) -> Dict[str,
         """
         pad_batch(batch, self.pad_value)
         return default_collate(batch)
+
+
+class ExtendDatasetSampler(Sampler):
+    """Samples elements randomly, while ensuring change in size of dataset.
+    Arguments:
+        ds_len : Length of original dataset.
+        ds_expand_len : Length to which original dataset must be expanded to.
+        shuffle : Whether to use shuffling.
+    """
+    def __init__(self, ds_len, ds_expand_len, shuffle=True):
+
+        self.ds_len = ds_len
+        self.ds_expand_len = ds_expand_len
+        self.shuffle = shuffle
+
+        self._check_input()
+
+        self.indices = []
+        self.base = [[i for i in range(self.ds_len)] for _ in range(math.ceil(self.ds_expand_len / self.ds_len))]
+
+    def __len__(self):
+        return self.ds_expand_len
+
+    def _check_input(self) -> None:
+        """Verify that the given input values are valid.
+        Raises:
+            AssertionError: If any of the parameters are found to by unacceptable for a variety of reasons.
+        """
+        assert isinstance(self.ds_expand_len, int), "Only accept positive integer type as ds_expand_len"
+        assert self.ds_expand_len > 0, "Invalid ds_expand_len. Expand Length cannot be less than or equal to 0"
+        assert isinstance(self.ds_len, int), "Only accept positive integer type as ds_len"
+        assert self.ds_len > 0, "Invalid ds_len. Dataset Length cannot be less than or equal to 0"
+
+    def __iter__(self):
+
+        self.indices = []
+        base = self.base
```

Comment:
this is at best a shallow copy, so won't protect the self.base from getting shuffled

Reply:
The shallow copy was done to protect the variable 'self.base' from getting sliced by the line below
![image](https://user-images.githubusercontent.com/44550569/142946171-200e2d9b-8415-4f05-bba3-47d15651459e.png)

Reply:
sure, but either this class is single use (in which case it doesn't need protection) or it is multi use (in which case you'd need to shield self.base from the shuffle)

Reply:
ah, I guess since shuffle can't be changed after init this isn't true. this is fine as is then